ee/tabby-db/src/conversions.rs

```diff
@@ -0,0 +1,28 @@
+pub trait DatabaseSerializable: Sized {
+    fn as_db_str(&self) -> &'static str;
+    fn from_db_str(s: &str) -> anyhow::Result<Self>;
+}
+
+#[macro_export]
+macro_rules! enum_mapping {
```

Comment:
1. if convinent, can this being implemented as an derivation macro? (e.g DbSerializable)
2. If not, I prefer making the traits implementation explicit at this stage, just for better readability

Reply:
I could certainly do that, we'd just have to define a new crate for it. Maybe we could make a `tabby-macros` crate?